### ISSUE TITLE
Why does assert False suggest pytest?

### ISSUE THREAD

**Issue by kaddkaka** [astral-sh/ruff#10143]
Upon linting a python file with only this content, ruff suggests to use pytest, why?

```py
assert False
```

`` ruff: Assertion always fails, replace with `pytest.fail()` [PT015] ``

There is nothing in filename that suggests that this is a test, so, why?

**Comment by mikeleppane** [comment]
Hi! If you enable this rule, it does not care about the file name or whether it is a normal source or a test file. You could specify and include PT rules only for test files. However, I agree, there may be room for improvement. We could try to figure out if it is a test file based on the conventions [Python test discovery](https://docs.pytest.org/en/7.2.x/explanation/goodpractices.html#conventions-for-python-test-discovery). I'm not 100% sure if there are some edge cases on top of this.

**Comment by kaddkaka** [comment]
Using standard pytest discovery seems reasonable. Including understanding pytest ini options such as

* `pyproject.toml`:

```toml
[tool.pytest.ini_options]
addopts = """\
--ignore=stupid_file.py"""
```

**Comment by MichaReiser** [comment]
For now, you can ignore the rule yourself by either ignoring the rule for all non test files using [`per-file-ignores`](https://docs.astral.sh/ruff/settings/#lint_per-file-ignores) or you can create a `ruff.toml` in your test directory (assuming all tests are in one directory), extend your root configuration by setting `extend = "../pyproject.toml"`, and enabling the pytest rules `extend-select = ["PT"]`

**Comment by charliermarsh** [comment]
I think this can be tracked in https://github.com/astral-sh/ruff/issues/8794.
